shared/src/sources/balancer/pool_storage.rs

```diff
@@ -81,6 +83,14 @@ impl PoolEvaluating for RegisteredWeightedPool {
     fn tokens(&self) -> Vec<H160> {
         self.common.tokens.clone()
     }
+
```

Comment:
One suggestion to make implementing this and require fewer methods:
```rust
trait PoolEvaluating {
   fn properties(&self) -> &CommonPoolInfo;
}
```

This should have the token list, address and scaling exponents right?

Reply:
Yes this is very nice. However, it is also important to note that Registered Pool (as it was before) is not quite the same as Balancer Pool that we are seeing here (while they are enums of various pool types, the PoolEvaluating will not cover both cases. We would need to rename `PoolEvaluating` to `RegisteredPoolEvaluating` and make yet another trait for doing the analogous thing for both Weighted and Stable Pools. The expected difference between these two traits is that one would have `CommonPoolInfo`(id, address, tokens, scaling_exponents) properties and the other would have (id, paused_state, swap_fee, reserves). 

Not that it should be a problem to do this, I just wanted to point this out in case you were thinking we would be able to reuse a lot.